lib/src/common/disposable.dart

```diff
@@ -26,19 +26,50 @@ abstract class _Disposable {
   Future<Null> dispose();
 }
 
-class InternalDisposable implements _Disposable {
+/// Used to invoke, and remove references to, a [Disposer] before disposal of the
+/// parent object.
+class ManagedDisposer implements _Disposable {
```

Comment:
I had searched through the code and thought we were only ever putting `ManagedDisposer`s in the `_internalDisposable` collection, but I must have missed the one case we're putting a `Disposable` in there. 🤷‍♂️